public/apps/account/log-out-button.tsx

```diff
@@ -17,28 +17,46 @@ import React from 'react';
 import { EuiButtonEmpty } from '@elastic/eui';
 import { HttpStart } from 'opensearch-dashboards/public';
 import { logout, samlLogout } from './utils';
+import { AuthType, OPENID_AUTH_LOGOUT } from '../../../common';
 
 export function LogoutButton(props: {
-  authType: string;
+  authType: string | string[];
   http: HttpStart;
   divider: JSX.Element;
   logoutUrl?: string;
 }) {
-  if (props.authType === 'openid') {
+  const currentAuthType = sessionStorage.getItem('current_auth_type');
+  let authArr = [];
+
+  if (typeof props.authType === 'string') {
+    authArr.push(props.authType);
+  } else {
+    authArr = [...props.authType];
+  }
+
+  if (
+    currentAuthType?.toLowerCase() === AuthType.OPEN_ID ||
```

Comment:
Can be taken away in a small function with `AuthType` as a parameter.

Reply:
Addressed.